### ISSUE TITLE
OSM: Support for Open Street Map

### ISSUE THREAD

**Issue by chanmratekoko** [primefaces/primefaces#3601]
Currently, Primefaces does not support OSM Map. 
GMap API is not free of charges. Web version API is limited for standard users. 
(Free up to 25,000 map loads per day.)
In this condition, Primefaces users have no choice for using other maps. 
OSM is free of charges. So, I think Primefaces need to support OSM at this time.

**Comment by tandraschko** [comment]
Feel free to provide a PR or sponsor it via PrimeFaces PRO.

**Comment by cnsgithub** [comment]
Btw, Google Maps API is unusable in China because it's blocked by the great firewall: https://en.greatfire.org/https/maps.googleapis.com

Whereas OpenStreetMap seems to be available: https://en.greatfire.org/https/openstreetmap.org

**Comment by kukel** [comment]
Anyone tried https://github.com/themrleon/JSF2Leaf ?

**Comment by chanmratekoko** [comment]
@kukel JSF2Leaf lib with together using Primefaces get entangled. Such as in p:wizard tag. After that, JSF2Leaf cannot be rendered properly with navigation-case forward.
Thanks!

**Comment by cnsgithub** [comment]
JSF2Leaf is licensed under GPLv2.

**Comment by melloware** [comment]
This is a duplicate of: https://github.com/primefaces/primefaces/issues/40

**Comment by mores** [comment]
BootsFaces does have an OpenStreetMap component: https://showcase.bootsfaces.net/forms/openStreetMap.jsf

And here is an article how to get BootsFaces and Primefaces to play well with each other: https://showcase.bootsfaces.net/integration/PrimeFaces.jsf

**Comment by melloware** [comment]
@mores you could always port the BootsFaces component to PrimeFaces Extensions if we really want OSM support.  I probably won't do it as I am a Google Maps / Azure Maps user at my clients.

**Comment by laurent-madarassou** [comment]
Hi everyone. 
Does OpenStreetMap component for BootsFaces support multi marker ?
Thanks a lot for answering.

**Comment by tandraschko** [comment]
better ask the bootsfaces guys for bootsfaces questions ;)

**Comment by mores** [comment]
> you could always port the BootsFaces component to PrimeFaces Extensions

Current work is in progress:
```
<h:panelGrid columns="4">
  <pe:osmap center="40.71264, -74.0154" zoom="13" style="width: 200px; height:200px" />
  <pe:osmap center="51.50673, -0.13526" zoom="13" style="width: 200px; height:200px" />
  <pe:osmap center="48.85724, 2.35012" zoom="13" style="width: 200px; height:200px" />
  <pe:osmap center="48.13507, 11.57726" zoom="13" style="width: 200px; height:200px" />
</h:panelGrid>```

<img width="847" alt="image" src="https://github.com/primefaces/primefaces/assets/1318377/01b86684-515a-4207-8bfe-5ac8333eb192">
